soc: microchip: sam: reorganize directory structure for sama7g5 series

TonyHan11 opened the pull request:
Update directory structure with following hierarchy for MPU devices:
```
  Product Architecture
    |__ SOC Series
          |__ Product Group
```
Add SAMA7G5 series System-in-Package (SiP) MPUs
Update makefiles, Kconfig symbols and header files accordingly

nordicjm reviewed (changes requested):
See changes in https://github.com/zephyrproject-rtos/zephyr/pull/97533 and fix PR

ArunMCHP commented on soc/microchip/sam/sama7/sama7g5/Kconfig.soc:
```diff
@@ -5,11 +5,11 @@
 	default "sama7g5" if SOC_SERIES_SAMA7G5
 
-rsource "Kconfig.soc.sam*"
+rsource "*/Kconfig.soc*"
```
```suggestion
config SOC_SAMA7G54
	bool
	select SOC_SERIES_SAMA7G5

config SOC_SAMA7G54D1G
	bool
	select SOC_SERIES_SAMA7G5

config SOC_SAMA7G54D1GN0
	bool
	select SOC_SERIES_SAMA7G5

config SOC_SAMA7G54D1GN2
	bool
	select SOC_SERIES_SAMA7G5

config SOC_SAMA7G54D2G
	bool
	select SOC_SERIES_SAMA7G5

config SOC_SAMA7G54D2GN4
	bool
	select SOC_SERIES_SAMA7G5

config SOC_SAMA7G54D4G
	bool
	select SOC_SERIES_SAMA7G5

config SOC
	default "sama7g54" if SOC_SAMA7G54
	default "sama7g54d1g" if SOC_SAMA7G54D1G
	default "sama7g54d1gn0" if SOC_SAMA7G54D1GN0
	default "sama7g54d1gn2" if SOC_SAMA7G54D1GN2
	default "sama7g54d2g" if SOC_SAMA7G54D2G
	default "sama7g54d2gn4" if SOC_SAMA7G54D2GN4
	default "sama7g54d4g" if SOC_SAMA7G54D4G
```

ArunMCHP commented on soc/microchip/sam/sama7/sama7g5/sama7g54/Kconfig.soc:
```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2025 Microchip Technology Inc. and its subsidiaries
```
You can remove this file and folder by adding these contents in `sama7/sama7g5/Kconfig.soc`

NhMchp reviewed (commented):
'mpu_clock' directory inside 'soc/microchip/sam' doesn't seems to be the correct approach. You should consider moving it to the clock driver. Another point is, the directory also has pinctrl files. So the name 'mpu_clock' also in incorrect. If other reviewers agree to keep this content, you can consider a new name 'mpu_common'.

JarmouniA commented:
> you can consider a new name 'mpu_common'.

Or just leave the original name, `common`.

TonyHan11 committed:
soc: microchip: sam: common: update CMakeLists.txt for more obvious

The 'common' folder is under 'soc/microchip/sam' which contains both MCU
and MPU devices. Currently all the files in 'common' is for MPU. Adding
if(foo) in 'CMakeLists.txt' to make more obvious of where the files are
used.
Update soc/microchip/sam/common/CMakeLists.txt with combining the .c
files with the same usage into one 'zephyr_source' call.

Signed-off-by: Tony Han <tony.han@microchip.com>

nordicjm commented:
I realise the PR Is wrong? The Kconfig **strings** are lowercase, the Kconfig **names** are uppercase

TonyHan11 committed:
soc: microchip: sam: reorganize directory structure for sama7g5 series

Update directory structure to the following hierarchy for MPU devices:
  Product Architecture
    |__ SOC Series
          |__ Product Group
Move directory 'soc/microchip/sama7g5/' into 'soc/microchip/sama7/'.
Add sama7g5 to 'soc/microchip/sam/sama7/soc.yml'.
Remove the files under soc/microchip/sam due to they became useless
with the reorganization.

Signed-off-by: Tony Han <tony.han@microchip.com>

TonyHan11 committed:
soc: microchip: sam: sama7: add sama7g54 (System-in-Package) SIP SoCs

Add SAMA7G5 series System-in-Package (SiP) MPUs to Kconfig.soc and
soc.yml, update the header files for them too.

Signed-off-by: Tony Han <tony.han@microchip.com>

TonyHan11 committed:
soc: microchip: sam: sama7: some optimize in Kconfig and CMakeLists.txt

Select MMU and CACHE_MANAGEMENT in 'config SOC_FAMILY_MICROCHIP_SAMA7'.
Replace 'config' with 'configdefault' for items in defconfig file.
Remove duplicated linker script in CMakeLists.txt in sama7d6 and sama7g5
directories, use the one in sama7 directory.

Signed-off-by: Tony Han <tony.han@microchip.com>

nordicjm commented on soc/microchip/sam/CMakeLists.txt:
squash into previous commit as above comment, this is not bisectable

TonyHan11 commented on soc/microchip/sam/sama7/sama7g5/Kconfig.soc:
```diff
@@ -5,9 +5,9 @@
 config SOC_SERIES_SAMA7G5
 	bool
-	select SOC_FAMILY_MICROCHIP_SAM
+	select SOC_FAMILY_MICROCHIP_SAMA7
```
Updated the first commit with squashing https://github.com/zephyrproject-rtos/zephyr/pull/97515/commits/2224b58259f70a12f534a68688ddd25abbdcf781 and updating `select SOC_FAMILY_MICROCHIP_SAMA7`, thanks a lot.

nordicjm commented on soc/microchip/sam/common/CMakeLists.txt:
```diff
@@ -3,16 +3,18 @@
-zephyr_sources(pmc.c)
-zephyr_sources_ifdef(CONFIG_SOC_SERIES_SAMA7D6 sama7d65.c)
-zephyr_sources_ifdef(CONFIG_SOC_SERIES_SAMA7G5 sama7g5.c)
+  zephyr_sources(clk-generated.c)
```
note: you can combine all of these into one function call e.g.
```suggestion
  zephyr_sources(
    clk-generated.c
    clk-main.c
    clk-master.c
    clk-peripheral.c
    clk-programmable.c
    clk-sam9x60-pll.c
    clk-system.c
    pmc.c
  )
  zephyr_sources_ifdef(CONFIG_SOC_SERIES_SAMA7G5
    clk-utmi.c
    sama7g5.c
  )
```

TonyHan11 replied:
> note: you can combine all of these into one function call e.g.

Updated, thanks.

nandojve commented:
ping @nordicjm @NhMchp @ArunMCHP
